tendermint/starport
# feat(module): bandchain oracle support

closes #1344
closes #1334

### What does this MR does?

Add possibility to request data from the Bandchain oracle.

### Changes

- Create oracle Bandchain template files;
- Add `band` CLI command;
- Create placeholders for oracle;
- Add Oracle modifiers;
- Add oracle integration tests;
- Add docs;

### How to test?

- Create and start your own oracle consumer cosmos blockchain (using only the --oracle flag automatically set the ibc flag to true):

```shell
$ starport s chain github.com/tendermint/bandchain
$ cd bandchain 
$ starport module create consuming --ibc
$ starport s band bandoracle --module consuming
```

- Configure and run the relayer:
```shell
$ starport relayer configure -a \
--source-rpc "http://rpc-laozi-testnet2.bandchain.org:26657" \
--source-faucet "https://laozi-testnet2.bandchain.org/faucet/request" \
--source-port "oracle" \
--source-gasprice "0uband" \
--source-gaslimit 5000000 \
--source-prefix "band" \
--source-version "bandchain-1" \
--target-rpc "http://localhost:26657" \
--target-faucet "http://localhost:4500" \
--target-port "consuming" \
--target-gasprice "0.0stake" \
--target-gaslimit 300000 \
--target-prefix "cosmos"  \
--target-version "bandchain-1"

$ starport relayer connect
```

- Request and verify data from the Bandchain blockchain:
```shell
$ bandchaind tx consuming bandoracle-data 37 4 3 --channel channel-0 --symbols "BTC,ETH,XRP,BCH" --multiplier 1000000 --fee-limit 30uband --request-key "random_string" --prepare-gas 600000 --execute-gas 600000 --from alice --chain-id bandchain

$ bandchaind query consuming last-bandoracle-id
request_id: "101276"

$ bandchaind query consuming bandoracle-result 101276
```
